torch_ttnn/passes/memory_pass.py

```diff
@@ -0,0 +1,233 @@
+                    # and if one of the users is a from device operation,
+                    # then swap out output tensor from device
+                    if not is_follow_up_tt_compute and is_follow_up_from_device:
+                        tid, output_size = self.deallocate(node)
+                        on_device_meta.remove((tid, output_size))
+
+                        self.mm.log(f"op removed from device: {node.name}\n")
+                        self.mm.log(f"output size: {output_size} bytes\n")
+                        self.mm.log(f"total SRAM usage: {self.mm.used_sram} bytes\n")
+
+                    # Once current node's input & output tensor processing is over, mark it as computed
+                    computed_nodes.append(node)
+                    # Swap out input tensors from device if its usage is completed
+                    # i.e there are no ttnn ops on device which uses the tensor
+                    for input_node in node.all_input_nodes:
+                        # Exception: Shape node is just like a parameter containing shape of input tensor
+                        if input_node.target == ttnn.Shape:
+                            continue
+                        keep_on_device = False
+                        input_node_users = list(input_node.users.keys())
+                        for user in input_node_users:
+                            if user not in computed_nodes and is_tt_compute(user):
+                                keep_on_device = True
+                        if keep_on_device is False:
+                            tid, tensor_size = self.deallocate(input_node)
+                            on_device_meta.remove((tid, tensor_size))
+
+                            self.mm.log(
+                                f"input tensor removed from device: {tensor_size} bytes\n"
+                            )
+                            self.mm.log(
+                                f"total SRAM usage: {self.mm.used_sram} bytes\n"
+                            )
+
+                    # data_points[(node.name, "from_device")] = on_device_meta.copy()
+
+        self.mm.set_peak_sram_usage()
+
+        self.mm.log(f"Tensor IDs to address map in SRAM:\n")
+        self.mm.log(f"{self.mm.tid_to_addr_map_in_sram}\n")
+        self.mm.log(
+            f"----------------------------------------------------------------\n"
+        )
+        self.mm.log(f"These ttnn ops overflow the SRAM buffer:\n")
+        for op in overflow_ops:
+            self.mm.log(f"{op}, ")
+
+        self.mm.log(f"\nData captured for plotting on a chart:\n")
+        for key, value in data_points.items():
+            self.mm.log(f"{key}: {value}\n")
+
+        # Convert dictionary keys to strings
+        data_str_keys = {str(key): value for key, value in data_points.items()}
+        self.mm.data_points = data_str_keys
+
+        if verbose:
+            print(self.mm.logs)
```

Comment:
Seems like if there's an exception before this, then the logs won't print. Would it be easier to debug the pass if the logs print as they are executed? If so, then the print can be incorporated into the `self.mm.log` member function.

Reply:
Yeah if there is a breakage somewhere, then difficult to debug without the logs. Yes print can be a part of log().